packages/core/src/ui/Base/TwitterEmbed.tsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useCallback, useState, useRef, RefObject } from "react";
+
+const twitterWidgetJs = "https://platform.twitter.com/widgets.js";
+const message = "Loading tweet...";
+
+interface TweetConfig {
+  theme: string;
+}
+
+type TweetState = {
+  isLoading: boolean;
+  message: string | null;
+};
+
+declare global {
+  interface Window {
+    twttr: {
+      widgets: {
+        createTweet: (
+          id: string,
+          ref: RefObject<HTMLDivElement>,
+          options: TweetConfig
+        ) => Promise<any>; // TODO type
+        load: (ref: RefObject<HTMLDivElement>) => void;
+      };
+    };
+  }
+}
+
+export default function TwitterEmbed({ url, ...props }) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [state, setState] = useState<TweetState>({
+    isLoading: true,
+    message: message,
+  });
+
+  const tweetId = url.split("status/").pop();
+
+  const renderTweet = useCallback(() => {
+    if (typeof window !== undefined) {
```

Comment:
I don't think you need this check, as you're only running `renderTweet` on the client.